crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -68,9 +71,44 @@ impl VisibleEntities {
     }
 }
 
+#[derive(Debug, Default, Clone)]
+pub struct EntityMeshRelationships {
+    entities_with_mesh: HashMap<Handle<Mesh>, SmallVec<[Entity; 1]>>,
+    mesh_for_entity: HashMap<Entity, Handle<Mesh>>,
+}
+
+impl EntityMeshRelationships {
+    /// Register the passed `entity` has having the passed `mesh_handle`.
+    ///
+    /// Note that this list _could_ have duplicates if an entity is assigned a new mesh and
+    /// then re-assigned the old mesh. This case should be rare so, for now, we'll risk
+    /// duplicating `Aabb` cloning+assigning.
+    fn register(&mut self, entity: Entity, mesh_handle: &Handle<Mesh>) {
+        self.entities_with_mesh
+            .entry(mesh_handle.clone_weak())
+            .or_default()
+            .push(entity);
+        self.mesh_for_entity
```

Comment:
In theory this `insert` should never return anything. Should we do anything if something _does_ come back out?